models/issue_label.go

```diff
@@ -257,6 +257,20 @@ func NewLabel(label *Label) error {
 	if !LabelColorPattern.MatchString(label.Color) {
 		return fmt.Errorf("bad color code: %s", label.Color)
 	}
+
+	// normalize case
+	label.Color = strings.ToLower(label.Color)
+
+	// add leading hash
+	if label.Color[0] != '#' {
+		label.Color = "#" + label.Color
+	}
+
+	// convert 3-character shorthand into 6-character version
+	if len(label.Color) == 4 {
+		label.Color = "#" + string(label.Color[1]) + string(label.Color[1]) + string(label.Color[2]) + string(label.Color[2]) + string(label.Color[3]) + string(label.Color[3])
```

Comment:
Rather than concat-ing these strings, could you use fmt.Sprintf instead?

The reason is that you don't need to cast anything to strings.

Reply:
Agree. Maybe something like:
```
r := color[1]
g := color[2]
b := color[3]
Sprintf("#%c%c%c%c%c%c", r, r, g, g, b, b)
```

Reply:
That does feel a lot cleaner; I'm not used to go, so, I'm not 100% sure what the best way to do things is.